loadbalancer_public.tf

```
@@ -243,7 +246,7 @@ resource "aws_lb_listener_certificate" "quortex_public" {
 
 resource "aws_lb_listener_rule" "quortex_public_tls_token_rule" {
   # This rule forwards traffic if the header token match the condition
-  count = var.load_balancer_public_expose_https ? 1 : 0
+  count = var.load_balancer_public_expose_https && length(var.load_balancer_public_app_backend_ports) > 0 ? 1 : 0
```

For consistency with your other changes : 
```suggestion
  count = length(var.load_balancer_public_app_backend_ports) > 0 && var.load_balancer_public_expose_https ? 1 : 0
```